sql/src/main/java/io/crate/operation/projectors/ShardingShardRequestAccumulator.java

```diff
@@ -196,25 +209,11 @@ public int batchSize() {
             TReq request = entry.getValue();
             it.remove();
 
-            ActionListener<ShardResponse> listener = new ActionListener<ShardResponse>() {
+            ShardId shardId = entry.getKey();
+            String nodeId = getDestinationNodeId(request, shardId);
```

Comment:
maybe we could change the grouping keys to a structure that contains both, the shardId and the nodeId to avoid having to re-retrieve the information?

This would also kind of fix the null-handling that currently results in warnings.